shell/common/rasterizer.cc

```diff
@@ -185,17 +222,19 @@ RasterStatus Rasterizer::Draw(
                  .GetRasterTaskRunner()
                  ->RunsTasksOnCurrentThread());
 
-  RasterStatus raster_status = RasterStatus::kFailed;
+  DoDrawResult draw_result;
   LayerTreePipeline::Consumer consumer =
-      [&](std::unique_ptr<LayerTreeItem> item) {
+      [this, &draw_result,
+       &discard_callback](std::unique_ptr<LayerTreeItem> item) {
         std::shared_ptr<LayerTree> layer_tree = std::move(item->layer_tree);
         std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder =
             std::move(item->frame_timings_recorder);
+        int64_t view_id = item->view_id;
         if (discard_callback(*layer_tree.get())) {
```

Comment:
I believe this discard callback is used when a window was resized. We discard the scene if its dimensions don't match the window's new size. We'll need to update this callback as each surface may have different dimensions.

Reply:
Sounds reasonable, I'll update it! Thank you.